src/bp-moderation/bp-moderation-filters.php

```diff
@@ -732,10 +756,233 @@ function bp_moderation_get_js_dependencies( $js_handles = array() ) {
+ *
+ * @return mixed Term meta data
+ */
+function bpm_category_show_when_reporting_column_display( $string = '', $column_name, $term_id ) {
+	$value = get_term_meta( $term_id, $column_name, true );
+	switch ( $value ) {
+		case 'members':
+			return esc_html__( 'Members', 'buddyboss' );
+		case 'content_members':
+			return esc_html__( 'Content & Members', 'buddyboss' );
+		default:
+			return esc_html__( 'Content', 'buddyboss' );
+	}
+}
+add_filter( 'manage_bpm_category_custom_column', 'bpm_category_show_when_reporting_column_display', 10, 3 );
+
+/**
+ * Display markup or template for custom field
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param string $column_name Column being shown.
+ * @param string $screen Post type being shown.
+ *
+ * @return mixed
+ */
+function bb_quick_edit_bpm_category_show_when_reporting_field( $column_name, $screen ) {
+	// If we're not iterating over our custom column, then skip.
+	if ( 'bpm_category' !== $screen && 'bpm_category_show_when_reporting' !== $column_name ) {
+		return false;
+	}
+	?>
+	<fieldset>
+		<div id="bpm_category_show_when_reporting" class="inline-edit-col">
+			<label>
+				<span class="title"><?php esc_html_e( 'Show When Reporting', 'buddyboss' ); ?></span>
+				<span class="input-text-wrap">
+					<select name="bpm_category_show_when_reporting" id="bpm_category_show_when_reporting">
+						<option value="content"><?php esc_html_e( 'Content', 'buddyboss' ); ?></option>
+						<option value="members"><?php esc_html_e( 'Members', 'buddyboss' ); ?></option>
+						<option value="content_members"><?php esc_html_e( 'Content & Members', 'buddyboss' ); ?></option>
+					</select>
+				</span>
+			</label>
+		</div>
+	</fieldset>
+	<?php
+}
+add_action( 'quick_edit_custom_box', 'bb_quick_edit_bpm_category_show_when_reporting_field', 10, 2 );
+
+
+/**
+ * Front-end stuff for pulling in user-input values dynamically into our input field.
+ *
+ * @since BuddyBoss [BBVERSION]
+ */
+function bb_quickedit_bpm_category_show_when_reporting_javascript() {
+	$current_screen = get_current_screen();
+
+	if ( 'edit-bpm_category' !== $current_screen->id || 'bpm_category' !== $current_screen->taxonomy ) {
+		return;
+	}
+
+	// Ensure jQuery library is loaded.
+	wp_enqueue_script( 'jquery' );
+	?>
+	<script type="text/javascript">
+		/*global jQuery*/
+		jQuery(function($) {
+			$('span:contains("Slug")').each(function (i) {
+				$(this).parent().remove();
+			});
+			$('#the-list').on('click', 'button.editinline', function(e) {
+				e.preventDefault();
+				var $tr = $(this).closest('tr');
+				var val = $tr.find('td.bpm_category_show_when_reporting').text();
+				if( val != '') {
+					$('tr.inline-edit-row select[name="bpm_category_show_when_reporting"] option').removeAttr('selected');
+					$('tr.inline-edit-row select[name="bpm_category_show_when_reporting"] option').filter(function() {
+						return this.text == val; 
+					}).attr('selected', 'selected');
+				}
+			});
+		});
+	</script>
```

Comment:
Can't we move this JS to the proper JS file?

Reply:
Moved the script to the settings page.js